lib/server/server.js

```diff
@@ -62,142 +64,269 @@ const {
   setServiceSources
 } = require('../service-data/service-data')
 
-const servicePath = path.resolve(process.env.PWD, SERVICE_PATH)
+const {
+  initControllers,
+  addControllers,
+  addModule
+} = require('../controller/controller')
+
+// data and schema resources to be loaded
+const dataSourceObjs = []
+const schemaObjs = []
+let componentDirs = []
+let servicePath
+let govukFrontendVersion
+
+const appDir = process.cwd()
+const kitLibDir = __dirname
+const kitDir = path.resolve(kitLibDir, '..')
+const cacheDir = path.join(appDir, ASSET_PATH, 'html')
+
+const prepareServiceSources = async () => {
+  if (!SERVICE_PATH) {
+    throw new FBServerError({
+      code: 'ENOSERVICEPATH',
+      message: 'No value for SERVICEPATH (path to service metadata) provided'
+    })
+  }
+  servicePath = path.resolve(process.env.PWD, SERVICE_PATH)
+  let govukFrontendPath
 
-const servicePackagePath = path.join(servicePath, 'package.json')
-const servicePackageLockPath = path.join(servicePath, 'package-lock.json')
-const serviceGitIgnorePath = path.join(servicePath, '.gitignore')
+  const servicePackagePath = path.join(servicePath, 'package.json')
+  const servicePackageLockPath = path.join(servicePath, 'package-lock.json')
+  const serviceGitIgnorePath = path.join(servicePath, '.gitignore')
+  const serviceNodeModulesPath = path.join(servicePath, 'node_modules')
 
-const servicePackageExists = fs.existsSync(servicePackagePath)
-if (!servicePackageExists) {
-  const defaultPackage = `
+  const servicePackageExists = fs.existsSync(servicePackagePath)
+  if (!servicePackageExists) {
+    const defaultPackage = `
 {
   "dependencies": {
     "@ministryofjustice/fb-components-core": "latest"
   }
 }
 `
-  fs.writeFileSync(servicePackagePath, defaultPackage)
-}
+    fs.writeFileSync(servicePackagePath, defaultPackage)
+  }
 
-if (!fs.existsSync(serviceGitIgnorePath)) {
-  const defaultGitIgnore = `
+  if (!fs.existsSync(serviceGitIgnorePath)) {
+    const defaultGitIgnore = `
 node_modules
 .DS_Store
 `
-  fs.writeFileSync(serviceGitIgnorePath, defaultGitIgnore)
-}
+    fs.writeFileSync(serviceGitIgnorePath, defaultGitIgnore)
+  }
 
-const cleanupPackage = !servicePackageExists ? `rm ${servicePackagePath} && rm ${servicePackageLockPath}` : ''
+  const cleanupPackage = !servicePackageExists ? `rm ${servicePackagePath} && rm ${servicePackageLockPath}` : ''
 
-execSync(`cd ${servicePath} && npm install`)
+  if (!PLATFORM_ENV) {
```

Comment:
If we're not in a production-style environment:

- check that there's an internet connection
- if yes and using latest
  -  delete any previously installed dependencies
- if no and no installed dependencies
  - spanner out

This will prevent getting stuck on a previous version of fb-components-core when running in the editor